Makefile

```diff
@@ -1,17 +1,24 @@
 OPENFAAS_URL?=http://127.0.0.1:8080/
 
-swarm:
-	docker service rm stronghash env-test env-test-labels env-test-annotations test-secret
-kubernetes:
+clean-swarm:
+	- docker service rm stronghash env-test env-test-labels env-test-annotations test-secret
+
+clean-kubernetes:
 	kubectl delete -n openfaas-fn deploy/stronghash ; kubectl delete -n openfaas-fn deploy/env-test
 	kubectl delete -n openfaas-fn env-test-labels ; kubectl delete -n openfaas-fn env-test-annotations
 	kubectl delete -n openfaas-fn deploy/test-secret
 
 SECRET?=tDsdf7sFT45gs8D3gDGhg54
 
 .EXPORT_ALL_VARIABLES:
-secrets:
+secrets-swarm:
 	./create-swarm-secret.sh
+
+secrets-kubernetes:
 	./create-kubernetes-secret.sh
-test: secrets
+
+test-swarm: secrets-swarm clean-swarm
```

Comment:
separate targets for testing different providers, if you are only running against `swarm` the `create-kubernetes-secret.sh` scripts hangs